### ISSUE TITLE
Error logging in for the first time - Error message in post below

### ISSUE THREAD

**Issue by johncarn** [opendocman/opendocman#214]
Just recently installed OpenDocMan v 1.3.3. When I try to access the web interface for the first time with Admin, i get the response:

You are here: Error
With a blue box below it that says
There was an error loggin you in. Please try again.    that blinks for a minute then stops. 

The environment I am running:
Windows Server 2012 Essentials 
Running the WampServer version of the pre-reqs:
- WampServer 2.5
- PHP 5.5.12
- mysql 5.6.17
- Apache server 2.4.9

I have tried with both Firefox and Internet Explorer and get the same error. 

I had to change the port that the system is running on to 8080. I updated the database where I could find as well as the apache server, etc. It seems to be ok because the URL on the page when I get the error is:
http://192.168.1.11:8080/error.php?ec=0

I checked the PHP, mySql and Apache logs and the only thing I see that is a little strange is a warning in the PHP log:
PHP Warning: PHP Startup: Unable to load dynamic library 'c:\wamp/bin/php/php5.5.12/php_intl.dll' - The specified module could not be found. in Unknown on line 0

I get the same message for php_ldap, but have verified that they are there.

I had various other issues getting things to work to this point, and had to make changes to the database entries (messed a couple up the first time I imported) but those errors have gone away.

Have already installed 
- Microsoft Visual C++ 2008 Redistributable - x64 v9.0.30729.6161
- Microsoft Visual C++ 2008 Redistributable - x86 v9.0.30729.4184
- Microsoft Visual C++ 2008 Redistributable - x86 v9.0.30729.6161
and
- Microsoft Visual C++ 2010 x64 Redistributable - v10.0.40219
- Microsoft Visual C++ 2012 Redistributable - x64 v11.0.61030
- Microsoft Visual C++ 2012 Redistributable - x64 v11.0.61030

I'm not sure where to look or what to try next. I seem to have all the pre-reqs in place. Have restarted all the services and get the same message. Only thing I haven't done is reboot since installation, but will try that this afternoon when I get time.

UPDATE: Rebooting the server did nothing. I still get the error.

Any suggestions?

Thanks.
jc

**Comment by stephenlawrence** [comment]
Check this first:
http://www.devside.net/wamp-server/unable-to-load-or-find-php-extension-php_intl-dll

**Comment by johncarn** [comment]
Thanks Stephen. Really appreciate the quick response and the link.

I did some searches but focused more on php instead of the wampserver. This is the first time I've used it. Seemed to be easier than installing all the components. Will give it a try.

Thanks again.
Cheers.

Thanks.
jc

Sent from my iPhone. Please excuse any typos or inaccurate auto-correct updates.

On Aug 11, 2015, at 3:51 PM, Stephen J. Lawrence Jr. <notifications@github.com<mailto:notifications@github.com>> wrote:


Check this first:
http://www.devside.net/wamp-server/unable-to-load-or-find-php-extension-php_intl-dll

-

**Comment by johncarn** [comment]
Hey Stephen,

Went through the link and corrected all the errors. I am not getting any errors in the mysql log, the php log, the apache log, or any of the windows log files.

Any other suggestions? I turned on developer on the browser and checked the debugger, but didn't see anything glaringly out of the ordinary. 

Thanks.
jc

**Comment by johncarn** [comment]
I found the problem tonight after a bunch of searching. I ran process monitor, I traced the file access, I spent several hours trying to troubleshoot this issue. Finally......

I went to the database.sql file to get the format to insert a new user and password to make sure I didn't have a problem there. When I started typing, I thought it looked a little funny, so I went and did a select * from odm_user. I was surprised to see that the Admin user had a password set. I didn't set it.

I went back to look at the docs and compare them to the database.sql file.

Manual Installation says to modify dataDir and base_url in the odm_settings table in the database.sql file. It says nothing about changing any values in the odm_user section.

Step 8 says Visit the URL for your installation and login as admin (no password).

Unfortunately, your installation instructions do not match the database.sql file provided in the installation for this release. In the Automate instructions it says to use the password you provided, but in the manual instructions it specifically states there is no password for the Admin user.

Step 8 under the manual installation section should state to login with user-id of admin, password of admin, unless you change the database.sql file you provide in the download.

Will be testing with it this week. Hopefully that's the last red herring I have to chase down <grin>
jc

**Comment by stephenlawrence** [comment]
Thanks JC,

I will update the docs.

**Comment by johncarn** [comment]
Hey Stephen,

Sorry if my notes came off harshly, I didn't mean it that way. It was late, I was tired, and I had been working for 19 hours straight when I finally figured it out. I was just trying to point out what was going on. If they were taken harshly, I truly apologize.

One thing I thought of after I sent the update is that the message I was getting was the same message I got before I fixed the PHP_intl.dll errors. Seems to me that if the password is wrong or missing for a user, they should get a message saying that password is missing or invalid, not just a generic error message. If error checking was on for the type of error it was, I would think it would be relatively easy to make that change. Had it been there, I would have resolved the issue much quicker.

So, enhancement request, maybe? Not sure how to do that with the forum.

jc

John Carney
President &
Chief Technology Therapist

484-228-8284

[cid:52C1BBA6-13D5-4339-8220-4DC59A3206ED]
